src/phpQuery/phpQueryObject.php

```diff
@@ -1,3180 +1,3584 @@
-//									? $node
-//									: null;
-//							else
-//								return ($index-$b)%$a == 0
-//									? $node
-//									: null;
-							'),
-						new CallbackParam(), $param
-					);
-				else
-					// index
-					$mapped = $this->map(
-						create_function('$node, $index',
-							'$prevs = pq($node)->prevAll()->size();
-							if ($prevs && $prevs == $index-1)
-								return $node;
-							else if (! $prevs && $index == 1)
-								return $node;
-							else
-								return null;'),
-						new CallbackParam(), $param
-					);
-				$this->elements = $mapped->elements;
-			break;
-			default:
-				$this->debug("Unknown pseudoclass '{$class}', skipping...");
-		}
-	}
-	/**
-	 * @access private
-	 */
-	protected function __pseudoClassParam($paramsString) {
-		// TODO;
-	}
-	/**
-	 * Enter description here...
-	 *
-	 * @return phpQueryObject|QueryTemplatesSource|QueryTemplatesParse|QueryTemplatesSourceQuery
-	 */
-	public function is($selector, $nodes = null) {
-		phpQuery::debug(array("Is:", $selector));
-		if (! $selector)
-			return false;
-		$oldStack = $this->elements;
-		$returnArray = false;
-		if ($nodes && is_array($nodes)) {
-			$this->elements = $nodes;
-		} else if ($nodes)
-			$this->elements = array($nodes);
-		$this->filter($selector, true);
-		$stack = $this->elements;
-		$this->elements = $oldStack;
-		if ($nodes)
-			return $stack ? $stack : null;
-		return (bool)count($stack);
-	}
-	/**
-	 * Enter description here...
-	 * jQuery difference.
-	 *
-	 * Callback:
-	 * - $index int
-	 * - $node DOMNode
-	 *
-	 * @return phpQueryObject|QueryTemplatesSource|QueryTemplatesParse|QueryTemplatesSourceQuery
-	 * @link http://docs.jquery.com/Traversing/filter
-	 */
-	public function filterCallback($callback, $_skipHistory = false) {
-		if (! $_skipHistory) {
-			$this->elementsBackup = $this->elements;
-			$this->debug("Filtering by callback");
-		}
-		$newStack = array();
-		foreach($this->elements as $index => $node) {
-			$result = phpQuery::callbackRun($callback, array($index, $node));
-			if (is_null($result) || (! is_null($result) && $result))
-				$newStack[] = $node;
-		}
-		$this->elements = $newStack;
-		return $_skipHistory
-			? $this
-			: $this->newInstance();
-	}
-	/**
-	 * Enter description here...
-	 *
-	 * @return phpQueryObject|QueryTemplatesSource|QueryTemplatesParse|QueryTemplatesSourceQuery
-	 * @link http://docs.jquery.com/Traversing/filter
-	 */
-	public function filter($selectors, $_skipHistory = false) {
-		if ($selectors instanceof Callback OR $selectors instanceof Closure)
-			return $this->filterCallback($selectors, $_skipHistory);
-		if (! $_skipHistory)
-			$this->elementsBackup = $this->elements;
-		$notSimpleSelector = array(' ', '>', '~', '+', '/');
-		if (! is_array($selectors))
-			$selectors = $this->parseSelector($selectors);
-		if (! $_skipHistory)
-			$this->debug(array("Filtering:", $selectors));
-		$finalStack = array();
-		foreach($selectors as $selector) {
-			$stack = array();
-			if (!
```

Comment:
can you revert this?

Reply:
Yes @lucassabreu, if this is going to be maintained

Reply:
ow, it marked too much

Reply:
ignore this for now

Reply:
Np

Reply:
well, there was a docblock here that i wanted to keep, but seens that was just some glitch on githubs diff, no changes needed

Reply:
So, no reverting needed?

Reply:
no reverting needed